### ISSUE TITLE
Retry on failed connect in Jenkins.getCrumb()

### ISSUE THREAD

**Issue by GreenGary** [KyleLNicholls/parameterized-builds#142]
We are suffering the following error on our site:
```
c.k.s.p.ciserver.Jenkins IOException in Jenkins.httpPost: Received close_notify during handshake
javax.net.ssl.SSLException: Received close_notify during handshake
```
It is rare, but frequent enough to be annoying (approx 2 times a week). 
The root cause for this may lie in the system's load, JRE flaws, OS flaws - google gives a lot of suggestions what to try to get rid of the problem. However, I cannot implement most of the suggestions due to company regulations.
It is not caused by a bug in the PlugIn, hence I would consider this issue an enhancement. The plug in may work around this issue by some retry mechanism. I implemented a first trial on a fork of this project and will update this issue as soon as I have collected more experience.

**Comment by a-hacker** [comment]
Hi @GreenGary,

Thank you for opening an issue for this. I'll make sure to take a look at any PR you open pertaining to this issue. If you have any questions, please feel free to ask me by commenting in this issue or on our gitter.

**Comment by GreenGary** [comment]
I have now two examples in the logs on our site, where the implemented retry helped to overcome the (temporary) connection problem.
I'll clean up my code and create a PR.

**Comment by a-hacker** [comment]
The feature has been released in version 3.2.6
